lib/sprockets.rb

```diff
@@ -162,6 +162,7 @@ module Sprockets
   require 'sprockets/eco_processor'
   require 'sprockets/ejs_processor'
   require 'sprockets/jst_processor'
+  register_mime_type 'application/javascript+function', extensions: []
```

Comment:
Yes, sprockets must know about all mime types or we will throw an exception as you mentioned. I don't know if this mime type declaration has any side effects, but I haven't seen any in the tests.